packages/frontend/src/providers/Dashboard/DashboardProvider.tsx

```diff
@@ -210,8 +210,8 @@ const DashboardProvider: React.FC<
         }
     }, [dashboard, dashboardFilters, overridesForSavedDashboardFilters]);
 
-    // Updates url with temp and overridden filters
-    useEffect(() => {
+    // Updates url with temp and overridden filters and deep compare to avoid unnecessary re-renders for dashboardTemporaryFilters
```

Comment:
dashboardTemporaryFilters were unstable even when there were no changes to them. 
After saving the dashboard, this useEffect would run with an outdated `pathname,` (edit pathname) in 245, which would do the following: user navigated back to view -> useeffect runs -> navigated back to edit